app/uk/gov/hmrc/agentsubscriptionfrontend/service/SessionStoreService.scala

```diff
@@ -72,6 +72,12 @@ class SessionStoreService @Inject()(sessionCache: SessionCache) {
   def fetchIsChangingAnswers(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Boolean]] =
     sessionCache.fetchAndGetEntry[Boolean]("isChangingAnswers")
 
+  def cacheAgentSession(agentSession: AgentSession)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Unit] =
+    sessionCache.cache("agentSession", agentSession).map(_ => ())
+
+  def fetchAgentSession(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[AgentSession]] =
+    sessionCache.fetchAndGetEntry[AgentSession]("agentSession")
```

Comment:
we should have a global case class for holding all data - we can not keep on adding stuff like this -